src/main/java/teammates/ui/pagedata/InstructorFeedbackResultsPageData.java

```diff
@@ -1168,14 +1206,21 @@ private void buildTableColumnHeaderForRecipientQuestionGiverView(List<ElementTag
             InstructorFeedbackResultsModerationButton moderationButton =
                     buildModerationButtonForExistingResponse(question, response);
 
+            boolean isStudentCommentsOnResponsesAllowed =
+                    question.getQuestionDetails().isStudentCommentsOnResponsesAllowed();
+
             InstructorFeedbackResultsResponseRow responseRow =
                     new InstructorFeedbackResultsResponseRow(
                             bundle.getGiverNameForResponse(response),
                             bundle.getTeamNameForEmail(response.giver),
                             bundle.getRecipientNameForResponse(response),
                             bundle.getTeamNameForEmail(response.recipient),
-                            bundle.getResponseAnswerHtml(response, question), moderationButton);
+                            bundle.getResponseAnswerHtml(response, question),
+                            moderationButton, isStudentCommentsOnResponsesAllowed);
 
+            if (isStudentCommentsOnResponsesAllowed) {
+                responseRow.setStudentComment(getResponseCommentText(response.giver, response));
```

Comment:
check if comment is empty

Reply:
What is the need to check if comment is empty? It will simply set studentComment to null.

Reply:
just a thought, if your comment is empty instead of null, in the frontend will there be a comment block that shown up be no text inside it? Just ensure that won't happen then I think this is fine